tests/serverpod_test_server/test_integration/serverpod_cli_args_override_test.dart

```diff
@@ -0,0 +1,342 @@
+      expect(serverpod.config.serverId, equals('cli-server'));
+      expect(serverpod.config.applyMigrations, isTrue);
+      // Config values (not overridden)
+      expect(serverpod.config.role, equals(ServerpodRole.monolith));
+      expect(serverpod.config.loggingMode, equals(ServerpodLoggingMode.normal));
+      expect(serverpod.config.applyRepairMigration, isFalse);
+    });
+
+    test(
+        'when no CLI args are provided and only config is given '
+        'then server uses config values except runMode which defaults to development',
+        () {
+      final config = ServerpodConfig(
+        runMode: 'staging',
+        serverId: 'config-server',
+        role: ServerpodRole.serverless,
+        loggingMode: ServerpodLoggingMode.verbose,
+        applyMigrations: true,
+        applyRepairMigration: true,
+        apiServer: ServerConfig(
+          port: 8080,
+          publicHost: 'localhost',
+          publicPort: 8080,
+          publicScheme: 'http',
+        ),
+        maxRequestSize: 524288,
+      );
+
+      final serverpod = Serverpod(
+        [],
+        Protocol(),
+        Endpoints(),
+        config: config,
+      );
+
+      // runMode defaults to 'development' when no CLI args or env vars are set
+      expect(serverpod.config.runMode, equals('development'));
+      // Other values should come from config
+      expect(serverpod.config.serverId, equals('config-server'));
+      expect(serverpod.config.role, equals(ServerpodRole.serverless));
+      expect(
+          serverpod.config.loggingMode, equals(ServerpodLoggingMode.verbose));
+      expect(serverpod.config.applyMigrations, isTrue);
+      expect(serverpod.config.applyRepairMigration, isTrue);
+    });
+
+    test(
+        'when invalid CLI args are provided with valid config '
+        'then server uses config values', () {
+      final config = ServerpodConfig(
+        runMode: 'development',
+        serverId: 'config-server',
+        role: ServerpodRole.monolith,
+        loggingMode: ServerpodLoggingMode.normal,
+        applyMigrations: true,
+        applyRepairMigration: false,
+        apiServer: ServerConfig(
+          port: 8080,
+          publicHost: 'localhost',
+          publicPort: 8080,
+          publicScheme: 'http',
+        ),
+        maxRequestSize: 524288,
+      );
+
+      final serverpod = Serverpod(
+        ['--invalid-option', 'value', '--another-invalid'],
+        Protocol(),
+        Endpoints(),
+        config: config,
+      );
+
+      // All values should come from config since CLI parsing fails
+      expect(serverpod.config.runMode, equals('development'));
+      expect(serverpod.config.serverId, equals('config-server'));
+      expect(serverpod.config.role, equals(ServerpodRole.monolith));
+      expect(serverpod.config.loggingMode, equals(ServerpodLoggingMode.normal));
+      expect(serverpod.config.applyMigrations, isTrue);
+      expect(serverpod.config.applyRepairMigration, isFalse);
+    });
```

Comment:
Suggestion: We don't necessarily need to check all the fields in this test, I suppose the reason for this test is to validate that we are not getting the default config instead of the Dart one?